CogniteSdk.FSharp/src/Assets.fs

```diff
@@ -0,0 +1,164 @@
+namespace CogniteSdk.FSharp
+
+open System
+open System.Collections.Generic
+
+open Oryx
+open Oryx.Cognite
+
+type AssetFilter =
+    { Name: string option
+      ParentExternalIds: string list
+      AssetSubtreeIds: Identifier list
+      DataSetIds: Identifier list
+      MetaData: Map<string, string>
+      Source: string option
+      CreatedTime: TimeRange option
+      LastUpdatedTime: TimeRange option
+      Root: bool option
+      ExternalIdPrefix: string option
+      Labels: LabelFilter option }
+
+    member x.ToAssetFilter() =
+        let filter = CogniteSdk.AssetFilter()
+
+        x.Name |> Option.iter (fun x -> filter.Name <- x)
+
+        x.ExternalIdPrefix
+        |> Option.iter (fun x -> filter.ExternalIdPrefix <- x)
+
+        x.Source
+        |> Option.iter (fun x -> filter.Source <- x)
+
+        x.Root |> Option.iter (fun x -> filter.Root <- x)
+
+        if not (List.isEmpty x.ParentExternalIds) then
+            filter.ParentExternalIds <- x.ParentExternalIds |> Array.ofList
+
+        if not (List.isEmpty x.AssetSubtreeIds) then
+            filter.AssetSubtreeIds <-
+                x.AssetSubtreeIds
+                |> List.map (fun x -> x.ToIdentifier())
+                |> Array.ofList
+
+        if not (List.isEmpty x.DataSetIds) then
+            filter.DataSetIds <-
+                x.DataSetIds
+                |> List.map (fun x -> x.ToIdentifier())
+                |> Array.ofList
+
+        x.Labels
+        |> Option.iter (fun x -> filter.Labels <- x.ToLabelFilter())
+
+        x.CreatedTime
+        |> Option.iter (fun x -> filter.CreatedTime <- x.ToTimeRange())
+
+        x.LastUpdatedTime
+        |> Option.iter (fun x -> filter.LastUpdatedTime <- x.ToTimeRange())
+
+        if not (Map.isEmpty x.MetaData) then
+            filter.Metadata <- x.MetaData |> Dictionary
+
+        filter
+
+    member x.Merge(other: AssetFilter option) : AssetFilter =
```

Comment:
The merge functions do not abide by union/intersection rules, as it uses union rules for lists (append operation) but option.orElse for single values. It might be safer to remove this than leave it for possible misuse.

Reply:
I still think this is what you want/need, but adding doc-strings to each method to describe what it's doing.

Reply:
This may be the case, just be aware that this function isn't complete if a dependent function uses this function to process and/or logic, since it's a mix of both 👍